File: 자료구조/족보/submit/hw/PP04/StackType.py
MAX_ITEMS = 100


class StackType:
    def __init__(self):
        self.info = []

    def is_empty(self):
        if len(self.info) == 0:
            return True
        return False

    def is_full(self):
        if len(self.info) == MAX_ITEMS:
            return True
        return False

    def push(self, item):
        if self.is_full() is False:
            self.info.append(item)

    def pop(self):
        if self.is_empty() is False:
            val = self.info.pop()
            return val

    def top(self):
        if self.is_empty() is False:
            return self.info[-1]
        return 'Stack is Empty'

File: 자료구조/족보/LAB/submit/2017100782_허진석_LAB11/ex04/StackType.cpp
#include "StackType.h"
#include <iostream>

StackType::StackType()
{
	pq = new PQType<StackNode>(100);
	length = 0;
}

StackType::~StackType()
{
	delete pq;
}

bool StackType::IsEmpty() const
{
	return pq->IsEmpty();
}

bool StackType::IsFull() const
{
	return pq->IsFull();
}

void StackType::Push(ItemType newItem)
{
	StackNode new_node;
	new_node.info = newItem;
	new_node.time_stamp = length++;
	pq->Enqueue(new_node);
}

void StackType::Pop()
{
	if (IsEmpty())
		throw EmptyStack();
	StackNode i;

	pq->Dequeue(i);
	length--;
}

ItemType StackType::Top()
{
	StackNode i;
	ItemType data;

	pq->Dequeue(i);
	data = i.info;
	pq->Enqueue(i);
	return data;
}
